### PR TITLE
fixes Special:SearchByProperty with reference datatype

### PR DESCRIPTION
**Pull request by thomas-topway-it**
- fixes https://github.com/SemanticMediaWiki/SemanticMediaWiki/issues/5474
by adding a `destructureDIContainer` method that retrieves the correct DIProperty and dataItem for the [reference](https://www.semantic-mediawiki.org/wiki/Help:Type_Reference) datatype (the first property of the multiValue dataItem) and passing them to `Store` -> `getPropertySubjects`

### CODE REVIEW

File: src/MediaWiki/Specials/SearchByProperty/QueryResultLookup.php [hunk 1]
```diff
@@ -207,14 +211,42 @@ private function doQueryForNonValue( PageRequestOptions $pageRequestOptions, Req
 			$requestOptions
 		);
 	}
+	
+	private function destructureDIContainer( $DIProperty, $dataItem, $pageRequestOptions ) {
```
Comment (JeroenDeDauw) [inline src/MediaWiki/Specials/SearchByProperty/QueryResultLookup.php:215]: Missing types

File: src/MediaWiki/Specials/SearchByProperty/QueryResultLookup.php [hunk 2]
```diff
@@ -207,14 +211,42 @@ private function doQueryForNonValue( PageRequestOptions $pageRequestOptions, Req
...
+		$multiValue = DataValueFactory::getInstance()->newDataValueByItem(
+			$dataItem,
+			$DIProperty
+		);
+		$properties = $multiValue->getPropertyDataItems( $property );
+
+		// Property for the reference value
+		$DIProperty = $properties[0];
```
Comment (JeroenDeDauw) [inline src/MediaWiki/Specials/SearchByProperty/QueryResultLookup.php:223]: Reusing a variable name like this isn't great

File: src/MediaWiki/Specials/SearchByProperty/QueryResultLookup.php (hunk 1 above)
Comment (thomas-topway-it) [reply src/MediaWiki/Specials/SearchByProperty/QueryResultLookup.php]: types added

File: src/MediaWiki/Specials/SearchByProperty/QueryResultLookup.php (hunk 2 above)
Comment (thomas-topway-it) [reply src/MediaWiki/Specials/SearchByProperty/QueryResultLookup.php]: I've renamed the variable and read the blog page. In this case I wasn't able to find a better way. I've added a comment with some explanation. An alternate way would be to create the correct dataitem manually using `DataValueFactory` however the class `PageRequestOptions` does some additional operation like unescaping the data value. I'm not sure if that can be avoided in this particular context.

**Pull request merged**